src/_P003_Pulse.ino

```diff
@@ -246,8 +247,9 @@ boolean Plugin_003(byte function, struct EventStruct *event, String& string)
       // task indexes larger than 32 should never happen
       if ( event->TaskIndex > sizeof(P003_initStepsFlags)*8 ) {
         // P003_initStepsFlags and P003_processingFlags can only serve as much tasks as their size in bits (32)
-        P003_log =  F("P003: Error! TaskIndex "); P003_log += event->TaskIndex; P003_log += F("is too large");
-        addLog(LOG_LEVEL_ERROR, P003_log);
+        String log;
+        log =  F("P003: Error! TaskIndex "); log += event->TaskIndex; log += F("is too large");
+        addLog(LOG_LEVEL_ERROR, log);
```

Comment:
Please wrap this in 
```c++
if (logLevelActiveFor(LOG_LEVEL_ERROR)) {
  String log;
  log.reserve((40);
  log = ....;
  log += ...;
  addLog(LOG_LEVEL_ERROR, log);
}
```

Reply:
I assume you mean "loglevelActiveFor" with lowercase "l".
Yes, that generates a better result. Disadvantage is, that code gets longer and diminishes readability.
I hope I caught now all places.
New version comes soon.